tests/ui/perfherder/alerts-view/status_dropdown_test.jsx

```diff
@@ -0,0 +1,124 @@
+import testPerformanceTags from '../../mock/performance_tags';
+import repos from '../../mock/repositories';
+import StatusDropdown from '../../../../ui/perfherder/alerts/StatusDropdown';
+
+const testAlertSummary = testAlertSummaries[0];
+const testAlerts = testAlertSummary.alerts;
+const testRepoModel = repos[2];
+
+const testUser = {
+  username: 'mozilla-ldap/test_user@mozilla.com',
+  isLoggedIn: true,
+  isStaff: true,
+  email: 'test_user@mozilla.com',
+};
+
+const testIssueTrackers = [
+  {
+    id: 1,
+    text: 'Bugzilla',
+    issueTrackerUrl: 'https://bugzilla.mozilla.org/show_bug.cgi?id=',
+  },
+  {
+    id: 2,
+    text: 'Github - Servo',
+    issueTrackerUrl: 'https://github.com/servo/servo/pull/',
+  },
+];
+
+const testStatusDropdown = (summaryTags) => {
+  testAlertSummary.performance_tags = summaryTags;
+
+  return render(
+    <StatusDropdown
+      alertSummary={testAlertSummary}
+      user={testUser}
+      updateState={() => {}}
+      repoModel={testRepoModel}
+      updateViewState={() => {}}
+      issueTrackers={testIssueTrackers}
+      bugTemplate={null}
+      filteredAlerts={testAlerts}
+      performanceTags={testPerformanceTags}
+    />,
+  );
+};
+
+afterEach(cleanup);
+
+test("Summary with no tags shows 'Add tags'", async () => {
+  const { getByText } = testStatusDropdown([]);
+
+  const dropdownItem = await waitFor(() => getByText('Add tags'));
+
+  expect(dropdownItem).toBeInTheDocument();
+});
+
+test("Summary with tags shows 'Edit tags'", async () => {
+  const { getByText } = testStatusDropdown(['harness']);
+
+  const dropdownItem = await waitFor(() => getByText('Edit tags'));
+
+  expect(dropdownItem).toBeInTheDocument();
+});
+
+test("Tags modal opens from 'Add tags'", async () => {
+  const { getByText, getByTestId } = testStatusDropdown([]);
+
+  const dropdownItem = await waitFor(() => getByText('Add tags'));
+
+  expect(dropdownItem).toBeInTheDocument();
+
+  fireEvent.click(dropdownItem);
+
+  const modal = await waitFor(() => getByTestId('tags-modal'));
+
+  expect(modal).toBeInTheDocument();
+});
+
+test("Tags modal opens from 'Edit tags'", async () => {
+  const { getByText, getByTestId } = testStatusDropdown(['harness']);
+
+  const dropdownItem = await waitFor(() => getByText('Edit tags'));
+
+  expect(dropdownItem).toBeInTheDocument();
+
+  fireEvent.click(dropdownItem);
+
+  const modal = await waitFor(() => getByTestId('tags-modal'));
+
+  expect(modal).toBeInTheDocument();
+});
+
+test("Tags modal closes from 'X'", async () => {
+  const { getByText, getByRole, getByTestId } = testStatusDropdown([]);
+
+  const dropdownItem = await waitFor(() => getByText('Add tags'));
+
+  expect(dropdownItem).toBeInTheDocument();
+
+  fireEvent.click(dropdownItem);
+
+  let modal = await waitFor(() => getByRole('dialog'));
+
+  expect(modal).toBeInTheDocument();
+
+  const header = await waitFor(() => getByTestId('tags-modal-header'));
+  const closeButton = header.lastChild;
```

Comment:
This makes our test dependent upon `reacstrap` implementation details. It could break from future updates of the 3rd party lib, which isn't desired.

Could you look for another way of getting a handle for the close button?